python/cuml/cuml/preprocessing/LabelEncoder.py

```diff
@@ -231,7 +233,9 @@ def fit_transform(self, y, z=None) -> cudf.Series:
         This is functionally equivalent to (but faster than)
         `LabelEncoder().fit(y).transform(y)`
         """
-
+        if cudf.get_option("mode.pandas_compatible"):
```

Comment:
cc: @vyasr @mroeschke @dantegd I'm debating whether this check should live inside `cudf.Series` constructor itself. I know I reverted the change in https://github.com/rapidsai/cudf/pull/17629/ but after looking at `cuml`'s frequency of usages of `cudf.Series/Index/DataFrame` constructors, I'm having second thoughts about a special utility(to check and extract true GPU object) in `cuml` vs baking this utility into `cudf` classics constructors.